src/core/types.ts

```diff
@@ -472,7 +472,7 @@ export interface MutationOptions<
   mutationFn?: MutationFunction<TData, TVariables>
   mutationKey?: MutationKey
   variables?: TVariables
-  onMutate?: (variables: TVariables) => Promise<TContext> | TContext
+  onMutate?: (variables: TVariables) => Promise<TContext> | TContext | undefined
```

Comment:
what about `Promise<undefined>` like in the docs?

Reply:
you mean like this???
```suggestion
  onMutate?: (variables: TVariables) => Promise<TContext | undefined> | TContext | undefined
```

Reply:
not sure if it isn't rather:

`Promise<TContext> | Promise<undefined> | TContext | undefined`

is there any difference 😅 ?

Reply:
i'm not sure. but your answer is good to me